modules/contact/src/main/kotlin/edu/stanford/spezi/modules/contact/model/Contact.kt

```diff
@@ -1,59 +1,30 @@
 package edu.stanford.spezi.modules.contact.model
 
+import android.location.Address
 import androidx.compose.ui.graphics.vector.ImageVector
 import java.util.UUID
 
 /**
  * Contact data class used to represent a contact.
  *
  * @param id the unique identifier of the contact
- * @param icon the icon of the contact
  * @param name the name of the contact
+ * @param image the image of the contact
  * @param title the title of the contact
  * @param description the description of the contact
  * @param organization the organization of the contact
  * @param address the address of the contact
  * @param options the list of contact options
  *
  * @see ContactOption
- * @see ContactOptionType
- *
  */
 data class Contact(
-    val id: UUID,
-    val icon: ImageVector?,
-    var name: String,
-    val title: String,
-    val description: String,
-    val organization: String,
-    val address: String,
+    val id: UUID = UUID.randomUUID(),
+    var name: PersonNameComponents,
+    val image: ImageVector?,
```

Comment:
Should we use new type to enable also local resources?

```kotlin
sealed interface ImageResource {
    data class Vector(val image: ImageVector) : ImageResource
    data class Drawable(@DrawableRes val resId: Int) : ImageResource
}
```

and depending on the type we use `Icon(image)` or `Icon(painter = painterResource(resId)`. If you like the approach, we can declare the new type in `design` module and create a custom icon composable taking that new type as an input. I see it to be used also in card options. Also, we can extend the data later on in case we want to support supplying an image url to be loaded asynchronously, e.g. via [coil](https://coil-kt.github.io/coil/compose/)

Reply:
I like the approach, yes! I'm not so sure about the Async part, since it involves more complexity (error handling, progress views, etc), but in general having some kind of union type for images is fine - I'm wondering whether this already exists in Jetpack Compose? Like `Image` in SwiftUI also allows to create images from multiple different origins (UIImage, asset names, system names, etc).

Reply:
this type doesn't exist yet, right?

Reply:
nope, there is no type yet and it should be created. and no, no compose component. Regarding the async part complexity that you mentioned a new custom type can also support that, e.g. `data class Async(val url: String, fallbackImage, loading, and so on)`, basically based on the use case we can think of custom needed configuration.

Reply:
I think this makes most sense to handle when addressing SpeziViews and I will implement this then. Adding a ticket for now, hope that's fine.

Reply:
https://github.com/StanfordSpezi/SpeziKt/issues/124